### DISCUSSION TITLE
Why isn't a rule to add/remove from piggy bank on withdrawal possilbe

### DISCUSSION THREAD

**Discussion by 7596ff** [firefly-iii/firefly-iii#4765] [Firefly III]
I'm trying to create rules that add a transaction amount to a piggy bank when I spend money on a credit card. I noticed in the logs that the rule isn't applying actions, because `[2021-05-10 13:53:21] local.INFO: Journal #6847 is a "Withdrawal" so skip this action.` What's the reason for this?

Additionally, why isn't there separate actions for add transaction amount or remove transaction amount? I want to add to the piggy bank (which saves on my checking account) on a withdrawal so that I can guarantee I have enough money for the purchase.

**Comment by JC5** [comment]
It's linked to the fact that piggy banks are generally used on savings accounts, where there are little to no withdrawals or deposits. 

What's your use case exactly for the second question?

**Reply by 7596ff** [reply]
I only want to spend money I actually have. So I try to keep enough money to instantly pay off my credit cards in my checking account if I need to. My rule would be whenever I spend money on a credit card, add that amount to the piggy bank associated with the card. The piggy bank is tied to my checking account so that the "Amount remaining for piggy banks" number is in effect the amount of money I can spend on a card before dipping into the negative.

**Reply by JC5** [reply]
Ah sorry no, that's something you'll have to set up using the API for example.
